Enhancement: Hability to reuse variable generate from template in dependencies for TestACC

from datasource vdc test
```
...
func (r *VDCDataSource) DependenciesConfig() (configs testsacc.TFData) {
	configs.Append(GetResourceConfig()[VDCResourceName]().GetDefaultConfig())
	return
}

func (r *VDCDataSource) Tests(ctx context.Context) map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test {
	return map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test{
		// * Test One (example)
		"example": func(_ context.Context, resourceName string) testsacc.Test {
			return testsacc.Test{
				// ! Create testing
				Create: testsacc.TFConfig{
					TFConfig: `
					data "cloudavenue_vdc" "example" {
						name = cloudavenue_vdc.example.name
					}`,
Checks: NewVDCResourceTest().Tests(ctx)["example"](ctx, resourceName).GenerateCheckWithCommonChecks(),
```

from resource vdc test 
```
				// ! Create testing
				Create: testsacc.TFConfig{
					TFConfig: testsacc.GenerateFromTemplate(resourceName, `
					resource "cloudavenue_vdc" "example" {
						name                  = {{ generate . "name" }}
						description           = {{ generate . "description" "longString"}}
						cpu_allocated         = 22000
						memory_allocated      = 30
						cpu_speed_in_mhz      = 2200
						billing_model         = "PAYG"
						disponibility_class   = "ONE-ROOM"
						service_class         = "STD"
						storage_billing_model = "PAYG"
					  
						storage_profiles = [{
						  class   = "gold"
						  default = true
						  limit   = 500
						}]
					  
					}`),
					Checks: []resource.TestCheckFunc{
						resource.TestCheckResourceAttr(resourceName, "name", testsacc.GetValueFromTemplate(resourceName, "name")),
						resource.TestCheckResourceAttr(resourceName, "description", testsacc.GetValueFromTemplate(resourceName, "description")),
						resource.TestCheckResourceAttr(resourceName, "cpu_allocated", "22000"),
						resource.TestCheckResourceAttr(resourceName, "memory_allocated", "30"),
						resource.TestCheckResourceAttr(resourceName, "cpu_speed_in_mhz", "2200"),
...
```

```
TF_ACC=1 go test  -v ./internal/testsacc/ -count=1 -timeout 30m -run TestVDCDataSource
=== RUN   TestVDCDataSource
    vdc_datasource_test.go:71: Step 1/1 error: Check failed: 2 errors occurred:
                * Check 9/13 error: data.cloudavenue_vdc.example: Attribute 'name' expected "QBWc9h64pGSvazlv", got "C3DZjzXsX2GaIyR0"
                * Check 10/13 error: data.cloudavenue_vdc.example: Attribute 'description' expected "Etiamne flete hi intuerer.", got "Pro conexos meo."
        
--- FAIL: TestVDCDataSource (57.62s)
FAIL
FAIL    github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/testsacc  57.988s
FAIL
```

--- a/internal/helpers/testsacc/template.go
+++ b/internal/helpers/testsacc/template.go
@@ -20,11 +20,11 @@ var (
 			}
 
 			randomString := generateRandomString(extraOpts[0])
-			(*KeyValueStore)[buildkeyValueStore(resourceName, key)] = randomString
+			(*KeyValueStore)[buildKeyValueStore(resourceName, key)] = randomString
 			return returnWithQuotes(randomString)
 		},
 		"get": func(resourceName, key string) string {
-			if v, ok := (*KeyValueStore)[buildkeyValueStore(resourceName, key)]; ok {
+			if v, ok := (*KeyValueStore)[buildKeyValueStore(resourceName, key)]; ok {
 				if s, ok := v.(string); ok {
 					return returnWithQuotes(s)
 				}
@@ -50,6 +50,9 @@ var (
 //   - generate: generates a random string and stores it in the key-value store. Generate accepts an optional argument that specifies the format of the random string (available formats: "shortString", "longString"). Default format is "shortString".
 //   - get: returns the value of the given key from the key-value store.
 func GenerateFromTemplate(resourceName, templateData string) TFData {
+	// if prefix of resourceName is "data." then remove it
+	resourceName = strings.TrimPrefix(resourceName, "data.")
+
 	t, _ := template.New(resourceName).Funcs(templateFuncs).Parse(templateData)
 	var tplTypes bytes.Buffer
 	_ = t.Execute(&tplTypes, resourceName)
@@ -60,16 +63,20 @@ func GenerateFromTemplate(resourceName, templateData string) TFData {
 
 // GetValueFromTemplate returns the value of the given key from the key-value store.
 func GetValueFromTemplate(resourceName, key string) string {
-	if v, ok := (*KeyValueStore)[buildkeyValueStore(resourceName, key)]; ok {
+	// if prefix of resourceName is "data." then remove it
+	resourceName = strings.TrimPrefix(resourceName, "data.")
+
+	if v, ok := (*KeyValueStore)[buildKeyValueStore(resourceName, key)]; ok {
 		if s, ok := v.(string); ok {
 			return s
 		}
 	}
+
 	return ""
 }
 
-// buildkeyValueStore builds the key-value store.
-func buildkeyValueStore(resourceName, key string) string {
+// buildKeyValueStore builds the key-value store.
+func buildKeyValueStore(resourceName, key string) string {
 	return resourceName + "." + key
 }
 

--- a/internal/helpers/testsacc/testsacc.go
+++ b/internal/helpers/testsacc/testsacc.go
@@ -303,6 +303,11 @@ func GenerateTests(tacc TestACC) []resource.TestStep {
 	return steps
 }
 
+// GenerateTestChecks Generate the checks for a specific test.
+func GenerateTestChecks(ctx context.Context, tacc TestACC, resourceName string, testName TestName) []resource.TestCheckFunc {
+	return tacc.Tests(ctx)[testName](ctx, resourceName).GenerateCheckWithCommonChecks()
+}
+
 // * Other
 
 // ImportStateIDBuilder is a function that can be used to dynamically generate

--- a/internal/testsacc/acctest.go
+++ b/internal/testsacc/acctest.go
@@ -12,11 +12,14 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/providerserver"
 
 	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 
 	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/helpers/testsacc"
 	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/metrics"
 	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/provider"
 )
+
+var localCacheResource = map[string]testsacc.Test{}
 
 // TestAccProtoV6ProviderFactories are used to instantiate a provider during
 // acceptance testing. The factory function will be invoked for every Terraform
@@ -82,11 +85,45 @@ func (r resourceConfig) GetDefaultConfig() testsacc.TFData {
 
 // GetSpecificConfig returns the create configuration for the test named "example".
 func (r resourceConfig) GetSpecificConfig(testName string) testsacc.TFData {
-	x := r.Tests(context.Background())[testsacc.TestName(testName)](
-		context.Background(),
-		r.GetResourceName(),
-	).Create.TFConfig
+	var x testsacc.TFData
+
+	if test, ok := localCacheResource[r.GetResourceName()+"."+testName]; ok {
+		x = test.Create.TFConfig
+	} else {
+		t := r.Tests(context.Background())[testsacc.TestName(testName)](
+			context.Background(),
+			r.GetResourceName()+"."+testName,
+		)
+		x = t.Create.TFConfig
+
+		localCacheResource[r.GetResourceName()+"."+testName] = t
+	}
+
 	x.AppendWithoutResourceName(r.DependenciesConfig())
+	return x
+}
+
+// GetDefaultChecks returns the checks for the test named "example".
+func (r resourceConfig) GetDefaultChecks() []resource.TestCheckFunc {
+	return r.GetSpecificChecks("example")
+}
+
+// GetSpecificChecks returns the checks for the test named.
+func (r resourceConfig) GetSpecificChecks(testName string) []resource.TestCheckFunc {
+	var x []resource.TestCheckFunc
+
+	if test, ok := localCacheResource[r.GetResourceName()+"."+testName]; ok {
+		x = test.Create.Checks
+	} else {
+		t := r.Tests(context.Background())[testsacc.TestName(testName)](
+			context.Background(),
+			r.GetResourceName()+"."+testName,
+		)
+		x = t.Create.Checks
+
+		localCacheResource[r.GetResourceName()+"."+testName] = t
+	}
+
 	return x
 }
 

--- a/internal/testsacc/vdc_datasource_test.go
+++ b/internal/testsacc/vdc_datasource_test.go
@@ -3,13 +3,11 @@ package testsacc
 
 import (
 	"context"
-	"regexp"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 
 	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/helpers/testsacc"
-	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/pkg/uuid"
 )
 
 var _ testsacc.TestACC = &VDCDataSource{}
@@ -37,7 +35,7 @@ func (r *VDCDataSource) DependenciesConfig() (configs testsacc.TFData) {
 func (r *VDCDataSource) Tests(ctx context.Context) map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test {
 	return map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test{
 		// * Test One (example)
-		"example": func(_ context.Context, resourceName string) testsacc.Test {
+		"example": func(_ context.Context, _ string) testsacc.Test {
 			return testsacc.Test{
 				// ! Create testing
 				Create: testsacc.TFConfig{
@@ -45,28 +43,14 @@ func (r *VDCDataSource) Tests(ctx context.Context) map[testsacc.TestName]func(ct
 					data "cloudavenue_vdc" "example" {
 						name = cloudavenue_vdc.example.name
 					}`,
-					// TODO: Bug to get value from template in dependencies issue #543
-					// Checks: NewVDCResourceTest().Tests(ctx)["example"](ctx, resourceName).GenerateCheckWithCommonChecks(),
-					Checks: []resource.TestCheckFunc{
-						resource.TestMatchResourceAttr(resourceName, "id", regexp.MustCompile(uuid.VDC.String()+`[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}`)),
-						resource.TestCheckResourceAttrSet(resourceName, "name"),
-						resource.TestCheckResourceAttrSet(resourceName, "description"),
-						resource.TestCheckResourceAttrSet(resourceName, "service_class"),
-						resource.TestCheckResourceAttrSet(resourceName, "disponibility_class"),
-						resource.TestCheckResourceAttrSet(resourceName, "billing_model"),
-						resource.TestCheckResourceAttrSet(resourceName, "cpu_speed_in_mhz"),
-						resource.TestCheckResourceAttrSet(resourceName, "cpu_allocated"),
-						resource.TestCheckResourceAttrSet(resourceName, "memory_allocated"),
-						resource.TestCheckResourceAttrSet(resourceName, "storage_billing_model"),
-						resource.TestCheckResourceAttr(resourceName, "storage_profiles.#", "1"),
-					},
+					Checks: GetResourceConfig()[VDCResourceName]().GetDefaultChecks(),
 				},
 			}
 		},
 	}
 }
 
-func TestVDCDataSource(t *testing.T) {
+func TestAccVDCDataSource(t *testing.T) {
 	resource.Test(t, resource.TestCase{
 		PreCheck:                 func() { TestAccPreCheck(t) },
 		ProtoV6ProviderFactories: TestAccProtoV6ProviderFactories,

